### PR TITLE
enable framework tests

### PR DESCRIPTION
**Pull request by hkollmann**
- add test action for framework with ubuntu, windows against node 10,12,14
- enable unhandledRejection in qx.Promise
- enable parameter handling in compile.js
- fix output format with own formatters
- fix unittests for firefox and ubuntu
- use own log function in qx.tool.util.Utils.print

result of testing could be seen here: https://github.com/hkollmann/qooxdoo/actions/runs/624874104.

### CODE REVIEW

File: .github/workflows/test-framework.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+name: run framework tests
+
+on:
+  push:
+    branches:
+      - '*'
+      - '!master'
```
Comment (derrell) [inline .github/workflows/test-framework.yml:7]: If we have tests run with master, wouldn't we be more likely to catch problems earlier?

File: source/class/qx/Promise.js [hunk 2]
```diff
@@ -903,7 +902,12 @@ qx.Class.define("qx.Promise", {
         return;
       }
       qx.Promise.__initialized = true;
-      qx.bom.Event.addNativeListener(window, "unhandledrejection", qx.Promise.__onUnhandledRejection.bind(this));
+      var isNode = typeof process !== "undefined";
+      if (isNode) {
+        process.on("unhandledRejection", qx.Promise.__onUnhandledRejection.bind(this));
```
Comment (derrell) [inline source/class/qx/Promise.js:907]: Maybe check whether the user has already registered an unhandled rejection handler? Or remove this one when they do register one?

**Review by derrell — CHANGES_REQUESTED** [review]

**Review by oetiker — APPROVED** [review]
cool ...

File: .github/workflows/test-framework.yml (hunk 1 above)
Comment (oetiker) [reply .github/workflows/test-framework.yml:7]: I think tests will run on the PR but on on master after the PR has been merged

File: source/class/qx/Promise.js (hunk 2 above)
Comment (hkollmann) [reply source/class/qx/Promise.js:907]: This is old framework code. Only adopted to node.

Comment (derrell) [reply source/class/qx/Promise.js:907]: Still an issue, but ok.

**Review by derrell — APPROVED** [review]

File: .github/workflows/test-framework.yml (hunk 1 above)
Comment (hkollmann) [reply .github/workflows/test-framework.yml:7]: Test will run on all branches except master. Therefore the !

Comment (derrell) [reply .github/workflows/test-framework.yml:7]: Yes I understand that. My question was, why shouldn't the tests run on master?

**Review by cboulanger — APPROVED** [review]
Thank you very much for the work involved!!

File: .github/workflows/test-framework.yml (hunk 1 above)
Comment (hkollmann) [reply .github/workflows/test-framework.yml:7]: ok. 2 reasons:
  - it's tested through the pull request before therefore not neccsessary
  - it's time consuming and prevents us for mass merges.
  -

Comment (derrell) [reply .github/workflows/test-framework.yml:7]: Thanks

**Comment by cboulanger** [comment]
Very clever new way of parsing the test command console output!

**Comment by hkollmann** [comment]
Based on idea of @oetiker

**Review by johnspackman — APPROVED** [review]

**Review by level420 — APPROVED** [review]

**Comment by cboulanger** [comment]
Thank you very much @hkollmann . Tests are BACK!!

**Pull request merged**